### PR TITLE
Add warnings and deprecate ackWarningsList and warningsList

### PR DESCRIPTION
**Pull request by jsom**

### CODE REVIEW

File: src/main/java/it/trade/model/reponse/TradeItPreviewStockOrEtfOrderResponse.java [hunk 1]
```diff
@@ -19,17 +15,32 @@
     @Expose
     public String orderId;
 
+    @SerializedName("warnings")
+    @Expose
+    public List<Warning> warnings = new ArrayList<>();
+
+    /**
+     * @deprecated Use warnings.
```
Comment (guillaumedebavelaere) [inline src/main/java/it/trade/model/reponse/TradeItPreviewStockOrEtfOrderResponse.java]: Can we get rid of this or is this still returned by the server?

Comment (jsom) [reply src/main/java/it/trade/model/reponse/TradeItPreviewStockOrEtfOrderResponse.java]: If we rip it out immediately anyone upgrading will see compilation errors.

File: src/main/java/it/trade/model/reponse/TradeItPreviewStockOrEtfOrderResponse.java [hunk 2]
```diff
@@ -19,17 +15,32 @@
     @Expose
     public String orderId;
 
+    @SerializedName("warnings")
```
Comment (guillaumedebavelaere) [inline src/main/java/it/trade/model/reponse/TradeItPreviewStockOrEtfOrderResponse.java]: From our documentation, it looks like it should be at the `orderDetails`level? (https://www.trade.it/documentation#PreviewTradeOrderDetails)

File: src/main/java/it/trade/model/reponse/TradeItPreviewStockOrEtfOrderResponse.java (hunk 1 above)
Comment (guillaumedebavelaere) [reply src/main/java/it/trade/model/reponse/TradeItPreviewStockOrEtfOrderResponse.java]: Yes, but we want them to update their code to use these new fields? Will they update their code if they don't see compilation errors?

File: src/main/java/it/trade/model/reponse/TradeItPreviewStockOrEtfOrderResponse.java (hunk 2 above)
Comment (jsom) [reply src/main/java/it/trade/model/reponse/TradeItPreviewStockOrEtfOrderResponse.java]: Good point. I thought they were at the same level as the others.

File: src/main/java/it/trade/model/reponse/TradeItPreviewStockOrEtfOrderResponse.java (hunk 1 above)
Comment (jsom) [reply src/main/java/it/trade/model/reponse/TradeItPreviewStockOrEtfOrderResponse.java]: They'll see a warning when compiling which should remind them. It's better to give integrators time to migrate as opposed to rip something out from underneath them.

**Pull request merged**